### DISCUSSION TITLE
Mocking jest connection for unit testing

### DISCUSSION THREAD

**Discussion by KhaledMAlasmari** [prisma/prisma#14078] [Q&A] [answered]
Hey, I've been trying to mock the DB connection for Postgresql but Prisma keeps connecting to the actual database despite me following this [guide](https://www.prisma.io/docs/guides/testing/unit-testing).
Here's my [repository ](https://github.com/KhaledMAlasmari/next.js-auth-prisma)
is there anything that I'm missing?
Thank you in advance

**Comment by austincrim** [comment, marked as answer]
Hey @KhaledMAlasmari 👋,

In your `jest.config.js`, change the instances of `<rootDir>` to your actual root directory, which is just `.` I believe. Also, since you're importing `singleton.js` in your `jest.setup.js`, you don't need to include it in your `setupFilesAfterEnv`.

```js
// jest.config.js
setupFilesAfterEnv: ['./jest.setup.js'],
```

**Reply by KhaledMAlasmari** [reply]
Thank you for your answer.
In the guide I mentioned above there is a step where you create a new file and export a new client instance in `client.ts`, I've been using that to mock the database. Apparently, that was the problem. I had to use the same client as the one in the function I'm trying to test.
I think there should be a clarification for headless chickens like me!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
